extension/src/workspace/gotoProvider.ts

```diff
@@ -0,0 +1,63 @@
+import * as vscode from 'vscode';
+import * as utils from '../utils';
+import { workspaceDocuments, activeDocuments } from "./workspaceInit";
+
+
+export class AutolispGoToProvider implements vscode.DefinitionProvider{
+	async provideDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.Location | vscode.Location[]> {
+		const editor: vscode.TextEditor = vscode.window.activeTextEditor;
+		let selected: string = editor.document.getText(editor.selection);	
+		// I tried using this 'addSelectionToNextFindMatch' trick, but it somehow not firing in time to trigger the event		
+		//			if (selected === "") {
+		//				await vscode.commands.executeCommand('editor.action.addSelectionToNextFindMatch');
+		//				selected = editor.document.getText(editor.selection);
+		//			}
+		// Settled on using a regex with a wide net and didn't use any (obvious) i18 vialoation characters 
+		if (selected === "") {
+			selected = editor.document.getText(editor.document.getWordRangeAtPosition(position, /\(\s*[\w\d\:\_\&\$\*\>\-]{1,99}/)).slice(1).trim();
+		}
+		return findGoToLocation(selected);
+	}
+}
+
+
+function findGoToLocation(searchFor: string): Promise<vscode.Location | vscode.Location[]> {
+	let kwRegex: RegExp = new RegExp("\\(\\s*(DEFUN|DEFUN-Q)\\s+" + utils.escapeRegExp(searchFor.toUpperCase()) + "(\\s|\\(|;)", "gi");
+	let found: Map<string, vscode.Position> = new Map();
+	// Try to find the definition in the currently opened documents
+	activeDocuments.forEach(doc =>{
```

Comment:
I think only `activeDocuments` is not enough, please think about the following case:
1. user open a .prj file in vscode Autolisp side bar
2. click a file in .prj tree to open a lisp file
3. go to definition 
Maybe this definition is in other lisp file referenced by this .prj file, in fact we EXPECT the `goto` feature can work for this case.

Reply:
Thanks for reminding me, I completely forgot about this context. Based on what you are saying, I think we can consider the PRJ references at the same precedence as truly Active documents. So, if you have no objection I'll lump those into the activeDocuments variable.